index.d.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import Store from '@orbit/store';
+import { TransformOrOperations, QueryBuilder, Source } from '@orbit/data';
+
+
+export interface DataProviderProps {
+  dataStore: Store;
+}
+
+export interface RecordsToProps {
+  [key: string]: (q: QueryBuilder) => any;
+}
+
+export interface WithData {
+  dataStore: Store;
+  sources: { [sourceName: string]: Source };
+}
+
+export type WithDataProps =
+  & {
+    queryStore: (transformOrOperations: TransformOrOperations, options?: object, id?: string) => any;
```

Comment:
Would it be possible to reference the upstream Store.query function here instead of copying its definition?

Reply:
Maybe! I'll give it a go